libc/src/__support/OSUtil/linux/quick_exit.cpp

```diff
@@ -22,14 +17,12 @@ namespace LIBC_NAMESPACE {
 #ifdef LIBC_TARGET_ARCH_IS_X86
 __attribute__((no_stack_protector))
 #endif
-LIBC_INLINE void
-quick_exit(int status) {
+__attribute__((noreturn))
+void quick_exit(int status) {
```

Comment:
Any way we could use C++ `[[]]` syntax for these?

Reply:
Actually, `[[gnu::no_stack_protector]]` was added to clang because of this.

Reply:
I see, adding attributes is fairly trivial, but unlike the GPU build we can't rely on an up-to-date compiler so I guess we'll just need to keep it as-is.

Reply:
yeah, I hit an issue with having these be repeated as separate attribute expressions, but I think I know how to fix that.

Reply:
done in https://github.com/llvm/llvm-project/pull/85955/commits/9ceeba4d5d7db1d58eeea33c31e46e9c5aea60a1 PTAL

Reply:
the issue was, clang errors if you mix GNU C `__attribute__`s with C23/C++20 `[[]]` attributes.

Reply:
[presubmit failure](https://buildkite.com/llvm-project/github-pull-requests/builds/49514#018e62f3-9c12-4483-9ee1-1f37a522324e)

> -- The C compiler identification is Clang 16.0.6
> llvm-project/github-pull-requests/libc/src/__support/OSUtil/linux/quick_exit.cpp:18:3: error: unknown attribute 'no_stack_protector' ignored [-Werror,-Wunknown-attributes]
[[gnu::no_stack_protector]]
  ^~~~~~~~~~~~~~~~~~~~~~~

So ironically, `no_stack_protector` was a function attribute in clang first, then GCC added support, then clang had to match the `gnu::` prefix later.  GCC does not recognize the `clang::` attribute prefix.  All the issues with compiler version compatibility are avoided by just using the GNU C extension syntax for attributes.

So I'm going to revert https://github.com/llvm/llvm-project/pull/85955/commits/9ceeba4d5d7db1d58eeea33c31e46e9c5aea60a1.

Reply:
Yeah, probably should've been more clear. This will work fine if you require runtimes builds of `clang` since it has the change, but not with the minimum clang version we support.